src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/missingnullcaseinswitch/InputMissingNullCaseInSwitchWithPrimitives.java

```diff
@@ -0,0 +1,87 @@
+/*
+MissingNullCaseInSwitch
+
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.coding.missingnullcaseinswitch;
+
+public class InputMissingNullCaseInSwitchWithPrimitives {
+    void testIntegers(int obj) {
+        switch (obj) {   // ok, case labels are primitive types
+            default -> {}
+            case 1 + 4 -> {}
+            case 2 -> {}
+        }
+        switch (obj) {
+           // case null -> {} // this will not compile
+            case 1 + 4 -> {}
+            case 2 -> {}
+            default -> {}
+        }
+        switch (obj) {
+            case 1 :{}
+            case 2: {}
+            default :{}
+        }
+         int w = switch (obj) {
+            case 1 + 2 + 5 + (( 1 + 3) + (3 + 1)) -> 2;
+            default -> 3;
+        };
+    }
+
+    void testChar(char c) {
+        switch (c) {   // ok, case labels are primitive types
+            default -> {}
+            case 'a' + 'c' -> {}
+            case 'b' -> {}
+        }
+        switch (c) {
+         // case null -> {} // this will not compile
+            case 'a' -> {}
+            case 'b' -> {}
+            default -> {}
+        }
+        switch (c) {
+            case 'a' :{}
+            case 'b': {}
+            default :{}
+        }
+    }
+    void testCaseLabelAsIdent(int obj) {
+        final int x = 1;
+        final int y = 0;
+        switch (obj) {   // ok, case labels are idents we ignore it
+            default -> {}
+            case x -> {}
+            case y -> {}
+        }
+        switch (obj) {
+           // case null -> {}
+            case x -> {}
+            case y -> {}
+            default -> {}
+        }
+        E e = E.X;
+        switch (e) {
+            case X :{}
+            case Y: {}
+        }
+    }
+
+    void boxedTypes(Integer i) {
+        switch (i) {   // case label are of a refetence type (Integer) but we can't detect
+            default -> {}
+            case 1 -> {}
+            case 2 -> {}
+        }
+
+        switch (i) {
+            case null -> {} // this compiles
```

Comment:
it works as intended, auto unboxing should occur when we encounter a case label that has an integer and the obj is not null. this may not be very accurate but this is as far as I understand.

Reply:
The bytecode is like: `invokedynamic #20,  0             // InvokeDynamic #0:typeSwitch:(Ljava/lang/Object;I)I`
It is then followed by a `tableSwitch` with a set of integers and addresses associated to those integers.

It takes the `i` as the first parameter, some constant 0 as the second parameter, and it returns an integer (I believe unrelated to what `i` really is).

IMO this is how I believe the bytecode works: the return value is the value to use in the later `tableSwitch` look up. `typeSwitch`'s basic logic is it first checks if `i` is null and if so returns -1. If that doesn't match, it then unboxes, and just returns the unboxed value as the result.

Reply:
It looks like the bytecode of a pattern case with a when is the same as if the user wrote out an if statement inside the case. It did do an auto-unbox when doing the comparison.

Interestingly it did a loop on the switch and increased the value of the second parameter to 1 to handle when the `when` didn't match.

Java code:
````
		Object obj = 0;

		switch (obj) {
		case Integer i when i == 0: {
			System.out.println("pattern");
		}
			break;
		case null: {
			System.out.println("null");
			break;
		}
		default: {
			System.out.println("default");
		}
		}
````
ChatGPT Java code from the bytecode:
````
    public static void main(String[] args) {
        Integer obj = Integer.valueOf(0);
        Integer temp = obj;
        int index = 0;

        while (true) {
            switch (typeSwitch(temp, index)) {
                case -1:
                    System.out.println("null");
                    break;
                case 0:
                    Integer intObj = (Integer) temp;
                    if (intObj.intValue() != 0) {
                        index = 1;
                        continue;
                    } else {
                        System.out.println("pattern");
                    }
                    break;
                default:
                    System.out.println("default");
                    break;
            }
            break;  // Exit the while loop
        }
    }

    public static int typeSwitch(Object obj, int index) {
        if (obj == null) {
            return -1; // Representing a null case
        }

        if (index == 0) {
            return 0; // Non-null, default behavior
        }

        // Other cases can be handled here if necessary
        return 1; // Default case
    }
````